### ISSUE TITLE
Rgb image with transparency color will have transparency removed when being saved

### ISSUE THREAD

**Issue by devedse** [SixLabors/ImageSharp#763]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have verified that I am running the latest version of ImageSharp
- [X] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [X] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

When simply opening and saving the image below all transparency is removed.

Input:
![snake](https://user-images.githubusercontent.com/2350015/47756148-3ca39900-dca1-11e8-88a1-c168db9f1dc7.png)

Output:
![snake_output](https://user-images.githubusercontent.com/2350015/47756174-53e28680-dca1-11e8-8570-971242eb4e50.png)

### Steps to Reproduce

The following code reproduces the problem:
```
using (var img = Image.Load(image1path))
{
    using (var fs = new FileStream(outputImage, FileMode.Create))
    {
        img.SaveAsPng(fs);
    }
}
``` 

### Investigation into issue

To help you guys out a bit I did some investigation myself to see where the issue comes from. First of all I did see that my image was detected as an Rgb image but it still had a transparent title bar in it.

PNG Analyzer shows a Transparency header though:
![image](https://user-images.githubusercontent.com/2350015/47756281-bdfb2b80-dca1-11e8-9099-089f9956b89d.png)

Doing some debugging I finally found out what was causing the problem. When the image is loaded a bool: 'hasTrans' is set to true and the transparent color is set to 211, 211, 211. This basically means that everywhere in the image where the color 211, 211, 211 is used it will be rendered as transparent.

However the issue occurs when we save the image again. The hasTrans flag is not taken into account when saving the image and thus it will simply be stored as Rgb without any data about transparency.

### Possible resolutions

As far as I can think of right now there's 2 possible resolutions:
1. Store transparency data on the image metadata and use this when writing the image again
2. If the data of an Rgb image contains a transparency value set to 0, save the image as Rgba32

I guess that both solution still have quite a bit of work to do though.

### System Configuration

- ImageSharp version: Github version as of 31-10-2018 and also Myget 1.0.0-dev002131
- Other ImageSharp packages and versions:
- Environment (Operating system, version and so on): Win10
- .NET Framework version: .NET Core 2.1
- Additional information:

**Comment by devedse** [comment]
Currently working on a Pull Request to implement this

**Comment by JimBobSquarePants** [comment]
Ah great stuff. I think I actually left a TODO in the code base regarding this.

**Comment by JimBobSquarePants** [comment]
Fixed with #764
